bundles/EcommerceFrameworkBundle/IndexService/Worker/AbstractBatchProcessingWorker.php

```diff
@@ -216,7 +231,24 @@ public function prepareDataForIndex(IndexableInterface $object)
                             $data[$attribute->getName()] = $this->convertArray($data[$attribute->getName()]);
                         }
                     } catch (\Exception $e) {
-                        Logger::err('Exception in IndexService: ' . $e);
+
+                        $event = new PreprocessAttributeErrorEvent($attribute, $e);
+                        $this->eventDispatcher->dispatch(IndexServiceEvents::ATTRIBUTE_PROCESSING_ERROR, $event);
+
+                        if ($event->isSkipAttribute()) {
+                            Logger::err(
+                                sprintf(
+                                    'Exception in IndexService when processing the attribute "%s": %s',
+                                    $event->getAttribute()->getName(),
+                                    $event->getException()->getMessage()
+                                )
+                            );
+                        } elseif ($event->isPropagationStopped()) {
```

Comment:
we should not use stopPropagation for this but add an additional flag on the event to decide if exception should be thrown or not